pkg/types/gcp/validation/machinepool.go

```diff
@@ -82,7 +83,7 @@ func ValidateDefaultDiskType(p *gcp.MachinePool, fldPath *field.Path) field.Erro
 	allErrs := field.ErrorList{}
 
 	if p != nil && p.OSDisk.DiskType != "" {
-		diskTypes := sets.NewString("pd-ssd")
+		diskTypes := sets.NewString("pd-ssd", "pd-balanced")
```

Comment:
The DefaultMachinePlatform provides a way to set values for both the compute and control plane machinepools. In order to prevent `pd-balanced` from being used with control-plane machines, you should probably not include this change.